libs/ml/examples/commodity_prediction/main.cpp

```diff
@@ -0,0 +1,349 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "file_loader.hpp"
+#include "math/tensor.hpp"
+#include "ml/dataloaders/tensor_dataloader.hpp"
+#include "ml/graph.hpp"
+#include "ml/layers/fully_connected.hpp"
+#include "ml/ops/activation.hpp"
+#include "ml/state_dict.hpp"
+
+#include <fstream>
+#include <iostream>
+#include <ml/ops/transpose.hpp>
+#include <string>
+
+using DataType  = double;
+using ArrayType = fetch::math::Tensor<DataType>;
+using GraphType = fetch::ml::Graph<ArrayType>;
+
+using namespace fetch::ml::ops;
+using namespace fetch::ml::layers;
+using namespace fetch::math;
+
+enum class LayerType
+{
+  NONE,
+  DENSE,
+  DROPOUT,
+  SOFTMAX
+};
+
+LayerType get_layer_type(std::string const &layer_name)
+{
+  LayerType layer_type    = LayerType::NONE;
+  int       match_counter = 0;
+
+  if (layer_name.find("dropout") != std::string::npos)
+  {
+    layer_type = LayerType::DROPOUT;
+    match_counter++;
+  }
+  if (layer_name.find("dense") != std::string::npos)
+  {
+    layer_type = LayerType::DENSE;
+    match_counter++;
+  }
+  if (layer_name.find("softmax") != std::string::npos)
+  {
+    layer_type = LayerType::SOFTMAX;
+    match_counter++;
+  }
+
+  assert(match_counter == 1);
+  return layer_type;
+}
+
+/**
+ * Loads a csv file into a Tensor
```

Comment:
presumably you're assuming a certain layout of the data (i.e. are rows == features or data?).
Should specify your format